src/ristretto/ristretto_keys.rs

```diff
@@ -74,29 +74,51 @@ impl borsh::BorshDeserialize for RistrettoSecretKey {
 //-----------------------------------------   Ristretto Secret Key    ------------------------------------------------//
 impl SecretKey for RistrettoSecretKey {
     const KEY_LEN: usize = 32;
+    const WIDE_REDUCTION_LEN: usize = 64;
 
     /// Return a random secret key on the `ristretto255` curve using the supplied CSPRNG.
     fn random<R: Rng + CryptoRng>(rng: &mut R) -> Self {
         RistrettoSecretKey(Scalar::random(rng))
     }
+
+    /// Return a secret key computed by wide reduction on a byte array
+    /// If the byte array is not exactly 64 bytes, returns an error
+    fn from_bytes_wide(bytes: &[u8]) -> Result<Self, ByteArrayError> {
+        if bytes.len() != Self::WIDE_REDUCTION_LEN {
+            return Err(ByteArrayError::IncorrectLength {});
+        }
+
+        // We need a slice of the correct length, which can't be enforced by the function signature
```

Comment:
I found this comment confusing. 

You have to copy because `Scalar::from_bytes_mod_order_wide` accepts a reference to an array of 64 bytes, whereas `bytes` is a reference to a byte slice.

Reply:
Right. The comment was intended to indicate that we can't have the function signature accept `bytes` as `&[u8; Self::WIDE_REDUCTION_LEN]` due to compiler restrictions, so the only solution is to do the copy. It wasn't explained well, so I just removed it.